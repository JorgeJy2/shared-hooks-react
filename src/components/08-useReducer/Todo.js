import React from 'react'
import PropTypes from 'prop-types';

export const Todo = ({ todo, index, handlerToggle, handlerDelete }) => {
    return (
        <li
            key={todo.id}
            className="list-group-item">
            <p className={`${todo.done && 'complete'}  `}
                onClick={() => { handlerToggle(todo.id) }}> {index + 1}. {todo.describe}
            </p>
            <button className="btn btn-outline-danger"
                onClick={() => { handlerDelete(todo.id) }} >
                x
        </button>
        </li>
    )
}

Todo.propTypes = {
    todo: PropTypes.object.isRequired,
    index: PropTypes.number.isRequired,
    handlerToggle: PropTypes.func.isRequired,
    handlerDelete: PropTypes.func.isRequired,
};