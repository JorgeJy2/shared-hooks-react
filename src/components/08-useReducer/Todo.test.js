import { shallow } from "enzyme";
import { Todo } from "./Todo";

const todos = [{
    id: 1,
    describe: 'Learn react',
    done: false
}, {
    id: 2,
    describe: 'Learn python',
    done: false
}];

describe('',() => {
    const handlerDelete = jest.fn(id => {});
    const handlerToggle = jest.fn(id => {});
    
    const wrapper = shallow(<Todo todo={todos[0]} index={1} handlerToggle={handlerToggle} handlerDelete={handlerDelete}/>);
    test('should snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should describe', () => {
        const describe = wrapper.find('p').text();
        expect(describe.includes(todos[0].describe)).toBeTruthy();
    });

    test('should delete', () => {
        wrapper.find('button').simulate('click');
        expect(handlerDelete).toBeCalledWith(todos[0].id);
    });

    test('should toggle', () => {
        wrapper.find('p').simulate('click');
        expect(handlerToggle).toBeCalledWith(todos[0].id);
    });

    test('should be class complete', () => {
        const todo = todos[0];
        todo.done = true;
        const wrapper = shallow(<Todo todo={todo} index={1} handlerToggle={handlerToggle} handlerDelete={handlerDelete}/>);
        expect(wrapper.find('p').hasClass('complete')).toBeTruthy();
    });
    
});