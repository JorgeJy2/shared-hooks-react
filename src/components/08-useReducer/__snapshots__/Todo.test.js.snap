// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` should snapshot 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="false  "
    onClick={[Function]}
  >
     
    2
    . 
    Learn react
  </p>
  <button
    className="btn btn-outline-danger"
    onClick={[Function]}
  >
    x
  </button>
</li>
`;
